### PR TITLE
feat: add `StudentT` distribution

### PR DESCRIPTION
**Pull request by ikrommyd**
Adds `StudentT` distribution

## Proposed Changes

- Implement `StudentT` from tensorflow probability

## Tests added

- Modify `tests/serialize/test_basics.py`  and add `tests/test_pdf_studentt.py`

## Checklist

- [x] change approved
- [x] implementation finished
- [x]  docs updated
- [x] correct namespace imported
- [x] tests added
- [x] CHANGELOG updated
- [x] (if new public method/class) docs in rst file updated?
- [x] (if large change) tutorial added/updated?

### CODE REVIEW

File: zfit/models/dist_tfp.py [hunk 1]
```diff
@@ -633,3 +633,76 @@ class ChiSquaredPDFRepr(BasePDFRepr):
...
+
+            f(x \\mid d, \\mu, \\sigma) = \\left(1 + \\frac{1}{d} \\left(\\frac{x - \\mu}{\\sigma}\\right)^2\\right)^{-\\frac{d+1}{2}} / Z
+
+        with the normalization over [-inf, inf] of
+
+        .. math::
+
+            Z = \\frac{\\Gamma(\\frac{d+1}{2})}{\\sqrt{\\pi d} \\Gamma(\\frac{d}{2})}
+
+        The normalization changes for different normalization ranges
```
Comment (jonas-eschle) [inline zfit/models/dist_tfp.py:665]:
```suggestion
        The effective normalization is given, as for all PDFs in `zfit`, through the `norm` argument that defaults to `obs`.
```

File: zfit/models/dist_tfp.py [hunk 2]
```diff
@@ -633,3 +633,76 @@ class ChiSquaredPDFRepr(BasePDFRepr):
...
+
+        Args:
+            ndof: Number of degrees of freedom
+            mu: Mean of the distribution
+            sigma: Scale of the distribution
+            obs: Observables and normalization range the pdf is defined in
```
Comment (jonas-eschle) [inline zfit/models/dist_tfp.py:671]:
```suggestion
            obs: |@doc:model.init.obs||@docend:model.init.obs|
```

File: zfit/models/dist_tfp.py [hunk 3]
```diff
@@ -633,3 +633,76 @@ class ChiSquaredPDFRepr(BasePDFRepr):
...
+               ``ext_*`` methods and the ``counts`` (for binned PDFs). |@docend:pdf.init.extended|
+            norm: |@doc:pdf.init.norm| Normalization of the PDF.
+               By default, this is the same as the default space of the PDF. |@docend:pdf.init.norm|
+            name: |@doc:model.init.name| Human-readable name
+               or label of
+               the PDF for better identification.
+               Has no programmatical functional purpose as identification. |@docend:model.init.name|
+        """
+        ndof, mu, sigma = self._check_input_params(ndof, mu, sigma)
+        params = OrderedDict((("ndof", ndof), ("mu", mu), ("sigma", sigma)))
```
Comment (jonas-eschle) [inline zfit/models/dist_tfp.py:685]:
```suggestion
        params = dict(ndof=ndof, mu=mu, sigma=sigma)
```
as dicts are insertion ordered, this should be all that is needed (I am also refactoring that for others)

File: CHANGELOG.rst [hunk 4]
```diff
@@ -21,6 +21,7 @@ Major Features and Improvements
 - add ``TruncatedPDF`` that allows to truncate in one or multiple ranges (replaces "MultipleLimits" and "MultiSpace")
 - add ``LogNormal`` PDF, a log-normal distribution, which is a normal distribution of the logarithm of the variable.
 - add ``ChiSquared`` PDF, the standard chi2 distribution, taken from [https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/Chi2](tensorflow-probability implementation).
+- add ``StudentT`` PDF, the standard Student's t distribution, taken from [https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/StudentT](tensorflow-probability implementation).
```
Comment (jonas-eschle) [inline CHANGELOG.rst:24]:
```suggestion
- add ``ChiSquared`` PDF, the standard chi2 distribution, taken from `tensorflow-probability implementation <https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/Chi2>`_.
- add ``StudentT`` PDF, the standard Student's t distribution, taken from `tensorflow-probability implementation <https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/StudentT>`_.
```
needs to be rst (yep, the chi2 was wrong)

**Review by jonas-eschle — COMMENTED** [review]
Looks good, indeed, straightforward. Minor comment

File: zfit/models/dist_tfp.py (hunk 2 above)
Comment (ikrommyd) [reply zfit/models/dist_tfp.py:671]: This is wrong in every tfp distribution

**Review by jonas-eschle — APPROVED** [review]
Looks all good to me, LGTM

**Review by jonas-eschle — APPROVED** [review]

**Pull request merged**